libs/perun/login/src/lib/login-screen/login-screen.component.ts

```diff
@@ -1,16 +1,29 @@
-import { Component } from '@angular/core';
-import { AuthService } from '@perun-web-apps/perun/services';
+import { Component, OnInit } from '@angular/core';
+import { AuthService, StoreService } from '@perun-web-apps/perun/services';
+import { ActivatedRoute } from '@angular/router';
+import { TranslateService } from '@ngx-translate/core';
 
 @Component({
   selector: 'perun-web-apps-login-screen',
   templateUrl: './login-screen.component.html',
   styleUrls: ['./login-screen.component.scss']
 })
-export class LoginScreenComponent {
+export class LoginScreenComponent implements OnInit{
 
-  constructor(
-    private auth: AuthService
-  ) { }
+  constructor(private auth: AuthService,
+              private route: ActivatedRoute,
+              private translateService: TranslateService,
+              private store: StoreService) { }
+
+  ngOnInit() {
+    this.route.queryParams.subscribe(params => {
+      const langs = this.store.get('supported_languages');
+      const urlLang = params['lang'];
+      if(urlLang && langs.includes(urlLang)) {
+        this.translateService.use(urlLang);
+      }
+    });
```

Comment:
and what about browser preferences? can we somehow get the browser preference for language?